tests/integration/devfile/cmd_devfile_describe_test.go

```diff
@@ -82,4 +84,35 @@ var _ = Describe("odo devfile describe command tests", func() {
 			helper.CmdShouldPass("odo", "delete", "-f", "--all", "--context", commonVar.Context)
 		})
 	})
+
+	Context("when running odo describe", func() {
+		It("should show json output for working cluster", func() {
+			helper.CmdShouldPass("odo", "create", "nodejs", "--context", commonVar.Context)
+			output := helper.CmdShouldPass("odo", "describe", "--context", commonVar.Context, "-o", "json")
+			values := gjson.GetMany(output, "kind", "metadata.name", "status.state")
+			Expect(helper.GjsonMatcher(values, []string{"Component", "nodejs", "Not Pushed"})).To(Equal(true))
+		})
+
+		It("should show json output for working cluster for a pushed component", func() {
+			newContext := path.Join(commonVar.Context, "newContext")
+			helper.MakeDir(newContext)
+			helper.Chdir(newContext)
```

Comment:
Why are we making and changing the directory here?

Reply:
because the top level context folder already contains a file which makes the `--starter` project download fail

Reply:
removed this spec as I used the one above it to check for this case as well